core/src/main/java/tc/oc/pgm/trigger/triggers/AbstractTrigger.java

```diff
@@ -0,0 +1,20 @@
+package tc.oc.pgm.trigger.triggers;
+
+import tc.oc.pgm.trigger.TriggerDefinition;
+
+public abstract class AbstractTrigger<S> implements TriggerDefinition<S> {
+
+  private final Class<S> cls;
+
+  public AbstractTrigger(Class<S> cls) {
+    this.cls = cls;
+  }
+
+  @Override
+  public Class<S> getScope() {
+    return cls;
+  }
+
+  @Override
+  public void untrigger(S s) {}
```

Comment:
Why not default both `trigger` and `untrigger`? It could make sense to keep the possibility open of a trigger only reacting to a filter falling instead of internally having to invert the filter response to use `trigger`

Reply:
The point of a default is just because *most triggers* will not have an `untrigger`, so you don't force them to add a method.
It is still perfectly possible for a trigger to implement untrigger and leave trigger empty, but usually you'd want them to implement trigger

Reply:
I agree using `trigger` should be the primary way of creating an Action, but I can't see a reason to just default implement `untrigger` preemptively?